### PR TITLE
Always keep cleaned elements

### PR DESCRIPTION
**Pull request by notriddle**
`keep_cleaned_elements` was added by users who were dissatisfied by how Ammonia behaved differently than other sanitizers (particularly the one used by GitHub). In practice, it seems like everybody turns it on, or doesn't care.

So, why do we have the other option again?

### CODE REVIEW

**Comment by lnicola** [comment]
Actually, that's my feature 😄. Actually, I wasn't aware what the other sanitizers do, but it seemed useful somewhat. I noticed today that @Eijebong also enables it in a project.

Note that the documentation still states that:

> Elements with invalid attributes are completely removed, to avoid confusion about what is and is not allowed.

That should be removed, too, in this PR.

File: src/lib.rs [hunk 1]
```diff
@@ -581,29 +566,6 @@ mod test {
...
-    }
-    #[test]
-    fn test_100() {
-        let fragment = "<!-- Here is a comment: -- This is a nested comment -->\n\
-                        <a href=\"http://harmless.com/This is still inside the comment: \
-                        --evadefilter><img onerror=alert(100) src=''/><a href=\"test\">\
-                        link</a>";
-        let result = clean(fragment);
-        assert_eq!(result, "\nlink");
-    }
```
Comment (lnicola) [inline src/lib.rs]: Are these no longer useful?

Comment (notriddle) [reply src/lib.rs]: They were never really useful. They tested the ability of the sanitizer to parse weird HTML corner cases, which means they're redundant to html5ever's own test suite. Using that linked test suite as designed requires you to involve a browser (the suite dosen't define what the resulting HTML should be, just that no JavaScript gets executed), which is really what we should do for that thing.

**Comment by notriddle** [comment]
Removed the outdated comment, and added this to the changelog.
